doc/MGMT_CLI.md

```diff
@@ -0,0 +1,420 @@
+Enable: True
+Port: 443
+token-refresh: 900
+token-validity: 3600
+
+#### 3.6.2.3 Exec Commands
+N/A
+
+### 3.6.3 REST API Support
+
+The sonic-mgmt-svcs YANG model will be available via the REST API as all other models are. The following URLs are provided.
+
+#### REST Server
+
+/restconf/data/sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST=default/port
+
+/restconf/data/sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST=default/client_auth
+
+/restconf/data/sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST=default/jwt_refresh
+
+/restconf/data/sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST=default/jwt_valid
+
+/restconf/data/sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST=default/enable
+
+
+#### Telemetry Server
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/port
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/client_auth
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/jwt_refresh
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/jwt_valid
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/enable
+
+
+### 3.6.4 gNMI Support
+
+gNMI Will support the same paths as REST but using the gNMI path/key format, i.e:
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/port
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/client_auth
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/jwt_refresh
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/jwt_valid
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/enable
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/port
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/client_auth
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/jwt_refresh
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/jwt_valid
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/enable
+
+
+
+## 3.7 Warm Boot Support
+N/A
+
+## 3.8 Upgrade and Downgrade Considerations
+
+The client_auth field will need to be migrated from the old format of a string with comma separated values to the new enum based field. A migrator function will be added to the migration script to handle this.
+
+The existing fields are a string with comma separated values (CSV). Since these values can be in any order, and the "none" option can invalidate the other modes, this must be considered during the migration.
+
+Below is a list of valid migrations:
+
+- password -> PASSWORD
+- password,jwt -> PASSWORD_JWT
+- none -> NONE
+- password,jwt,cert -> PASSWORD_JWT_CERT
+- password,cert -> PASSWORD_CERT
+- jwt,cert -> JWT_CERT
+- cert -> CERT
+
```

Comment:
Can we retain the comma separated list? This is a standard way of representing array in sonic db. It can be modeled as a leaf-list in sonic yang. We may have to change the field name as "client_auth@" to match swsssdk conventions. DB migration script needs to change the field name only.